app/routes/titiler/umd_glad_dist_alerts.py

```diff
@@ -49,24 +50,61 @@ async def glad_dist_alerts_raster_tile(
         AlertConfidence.low,
         description="Show alerts that are at least of this confidence level",
     ),
+    tree_cover_density: Optional[int] = Query(
+        None,
+        ge=0,
+        le=100,
+        description="Alerts in pixels with tree cover density (in percent) below this threshold won't be displayed. `umd_tree_cover_density_2010` is used for this masking.",
+    ),
+    tree_cover_height: Optional[int] = Query(
+        None,
+        description="Alerts in pixels with tree cover height (in meters) below this threshold won't be displayed. `umd_tree_cover_height_2020` dataset in the API is used for this masking.",
+    ),
+    tree_cover_loss_cutoff: bool = Query(
+        False,
+        ge=2021,
+        description="""This filter is to be used on conjunction with `tree_cover_density` and `tree_cover_height` filters to detect only alerts in forests by masking out pixels that have had tree cover loss prior to the alert.""",
+    ),
 ) -> Response:
     """UMD GLAD DIST alerts raster tiles."""
 
+    tile_x, tile_y, zoom = xyz
     bands = ["default", "intensity"]
     folder: str = f"s3://{DATA_LAKE_BUCKET}/{dataset}/{version}/raster/epsg-4326/cog"
     with AlertsReader(input=folder) as reader:
-        tile_x, tile_y, zoom = xyz
-
         # NOTE: the bands in the output `image_data` array will be in the order of
         # the input `bands` list
         image_data = reader.tile(tile_x, tile_y, zoom, bands=bands)
 
-    processed_image = DISTAlerts(
+    dist_alert = DISTAlerts(
         start_date=start_date,
         end_date=end_date,
         render_type=render_type,
         alert_confidence=alert_confidence,
-    )(image_data)
+        tree_cover_density_mask=tree_cover_density,
+        tree_cover_height_mask=tree_cover_height,
+        tree_cover_loss_mask=tree_cover_loss_cutoff,
+    )
+
+    if tree_cover_density:
+        with COGReader(
+            f"s3://{DATA_LAKE_BUCKET}/umd_tree_cover_density_2010/v1.6/raster/epsg-4326/cog/default.tif"
+        ) as reader:
+            dist_alert.tree_cover_density_data = reader.tile(tile_x, tile_y, zoom)
+
+    if tree_cover_height:
+        with COGReader(
+            f"s3://{DATA_LAKE_BUCKET}/umd_tree_cover_height_2020/v2022/raster/epsg-4326/cog/default.tif"
+        ) as reader:
+            dist_alert.tree_cover_height_data = reader.tile(tile_x, tile_y, zoom)
+
+    if tree_cover_loss_cutoff:
+        with COGReader(
+            f"s3://{DATA_LAKE_BUCKET}/umd_tree_cover_loss/v1.10.1/raster/epsg-4326/cog/default.tif"
```

Comment:
Put in a TODO here to fix the version name (to v1.11) once we go from staging to production?

Reply:
How often will we need to change the code? Should this be a config variable somewhere?

Reply:
@gtempus: I hit what's potentially a terraform bug setting an env variable that's a json string with the dataset versions - same variable that works fine with the docker-compose file. So, for now I have moved out the dataset versions into the global config file where we'll have to update it (https://github.com/wri/gfw-tile-cache/pull/170/commits/693e0c243b459fc635f01465d8df5b82c3f7b90c) opened and an issue in terraform aws provider that I hope will get sorted out.

Reply:
This is a good interim solution, @solomon-negusse. Thanks! 
It's much better than having to dig into the codebase. 😄 
Bummer about the terraform bug. 😦